### PR TITLE
Issue 14/preference page

### LINKED ISSUE #14: Add an eclipse preference page for Skeditor
The page should feature a component (e.g. radio button) to select z3 or Mathematica as prover and a component to specify the corresponding directory.

### PR DESCRIPTION
**Pull request by pumpkinchariot**

### CODE REVIEW

File: Plugins/de.tubs.skeditor/.classpath [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+	<classpathentry kind="src" path="src-gen/de/tubs/skeditor"/>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8">
+		<attributes>
+			<attribute name="module" value="true"/>
+		</attributes>
+	</classpathentry>
+	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
+	<classpathentry kind="lib" path="libs/keymaerax.jar" sourcepath="libs/keymaerax.jar"/>
+	<classpathentry kind="lib" path="C:/GitHub/Skeditor/Plugins/de.tubs.skeditor/libs/com.microsoft.z3.jar"/>
```
Comment (AlexanderKnueppel) [inline Plugins/de.tubs.skeditor/.classpath:12]: Add relative path

File: Plugins/de.tubs.skeditor/src/de/tubs/skeditor/keymaera/KeYmaeraBridge.java [hunk 2]
```diff
@@ -42,39 +42,40 @@ public static Sequent parseSeqFromString(String content) {
...
 		return JavaConverters$.MODULE$.mapAsScalaMapConverter(m).asScala()
 				.toMap(scala.Predef$.MODULE$.<scala.Tuple2<A, B>>$conforms());
 	}
-	
+
 	public static java.util.HashMap<String, String> getZ3Config() {
 		java.util.HashMap<String, String> c = new HashMap<String, String>();
-		c.put("z3Path", "C:\\Users\\Dibo\\.keymaerax\\z3.exe"); // path
+		c.put("z3Path", SkeditorSettings.getInstance().getZ3Exe());
+		// c.put("z3Path", "C:\\Users\\Dibo\\.keymaerax\\z3.exe"); // path
```
Comment (AlexanderKnueppel) [inline Plugins/de.tubs.skeditor/src/de/tubs/skeditor/keymaera/KeYmaeraBridge.java:65]: Remove comment

File: Plugins/de.tubs.skeditor/src/de/tubs/skeditor/keymaera/KeYmaeraBridge.java [hunk 3]
```diff
@@ -42,39 +42,40 @@ public static Sequent parseSeqFromString(String content) {
...
-		c.put("linkName", "C:\\Program Files\\Wolfram Research\\Mathematica\\11.2\\MathKernel.exe"); // path
-		c.put("libDir",
-				"C:\\Program Files\\Wolfram Research\\Mathematica\\11.2\\SystemFiles\\Links\\JLink\\SystemFiles\\Libraries\\Windows-x86-64\\"); // path
+
+		c.put("linkName", SkeditorSettings.getInstance().getMathematicaExe());
+		c.put("libDir", SkeditorSettings.getInstance().getMathematicaLibs());
+
+//		c.put("linkName", "C:\\Program Files\\Wolfram Research\\Mathematica\\11.2\\MathKernel.exe"); // path
+//		c.put("libDir",
+//				"C:\\Program Files\\Wolfram Research\\Mathematica\\11.2\\SystemFiles\\Links\\JLink\\SystemFiles\\Libraries\\Windows-x86-64\\"); // path
```
Comment (AlexanderKnueppel) [inline Plugins/de.tubs.skeditor/src/de/tubs/skeditor/keymaera/KeYmaeraBridge.java:53]: Remove comment

File: Plugins/de.tubs.skeditor/src/de/tubs/skeditor/preferences/SelectProverGroup.java [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+		RadioButtonAndFileBrowserComposite z3Composite = new RadioButtonAndFileBrowserComposite(this, 0, "z3");
+		z3Composite.pack();
+
+		RadioButtonAndFileBrowserComposite mathematicaComposite = new RadioButtonAndFileBrowserComposite(this, 0,
+				"mathematica");
+		mathematicaComposite.pack();
+
+//		RadioButtonAndFileBrowserComposite secondMathematicaComposite = new RadioButtonAndFileBrowserComposite(this, 0,
+//				"");
+//		secondMathematicaComposite.pack();
```
Comment (AlexanderKnueppel) [inline Plugins/de.tubs.skeditor/src/de/tubs/skeditor/preferences/SelectProverGroup.java:56]: Remove comment

File: Plugins/de.tubs.skeditor/src/de/tubs/skeditor/preferences/SelectProverGroup.java [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+			}
+
+			Text directoryText = new Text(this, SWT.BORDER);
+			RowData textLayoutData = new RowData();
+			textLayoutData.width = 200;
+			directoryText.setLayoutData(textLayoutData);
+			directoryText.addModifyListener(new ModifyListener() {
+				@Override
+				public void modifyText(ModifyEvent e) {
+					//System.out.println("TEST " + ((Text) e.widget).getText());
```
Comment (AlexanderKnueppel) [inline Plugins/de.tubs.skeditor/src/de/tubs/skeditor/preferences/SelectProverGroup.java:130]: Remove

File: Plugins/de.tubs.skeditor/src/de/tubs/skeditor/preferences/FileDialogSelectionListener.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+ * @author Dibo Gonda
+ *
+ */
+public class FileDialogSelectionListener extends SelectionAdapter {
+
+	private final Shell shell;
+	private final Text target;
+	private static final String[] FILTER_NAMES = { "Executable (*.exe)" };
+
+	private static final String[] FILTER_EXTS = { "*.exe" };
```
Comment (AlexanderKnueppel) [inline Plugins/de.tubs.skeditor/src/de/tubs/skeditor/preferences/FileDialogSelectionListener.java:23]: Binary

File: Plugins/de.tubs.skeditor/src/de/tubs/skeditor/keymaera/KeYmaeraBridge.java [hunk 7]
```diff
@@ -42,39 +42,40 @@ public static Sequent parseSeqFromString(String content) {
 		return edu.cmu.cs.ls.keymaerax.parser.StringConverter$.MODULE$.StringToStringConverter(content).asSequent();
```
Comment (aimiebell) [inline Plugins/de.tubs.skeditor/src/de/tubs/skeditor/keymaera/KeYmaeraBridge.java:42]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

**Pull request merged**